pkg/controller/machinepool/azureactuator.go

```diff
@@ -151,3 +166,32 @@ func (a *AzureActuator) getZones(region string, instanceType string) ([]string,
 
 	return nil, err
 }
+
+func (a *AzureActuator) getImagesByResourceGroup(resourceGroupName string) ([]string, error) {
+	ctx, cancel := context.WithTimeout(context.TODO(), 30*time.Second)
+	defer cancel()
+
+	var images []string
+	var res azureclient.ImageListResultPage
+	var err error
+	for res, err = a.client.ListImagesByResourceGroup(ctx, resourceGroupName); err == nil && res.NotDone(); err = res.NextWithContext(ctx) {
+		for _, image := range res.Values() {
+			images = append(images, *image.Name)
+		}
+	}
+
+	return images, err
+}
+
+func (a *AzureActuator) gen2ImageExists(infraID, resourceGroupName string) (bool, error) {
+	images, err := a.getImagesByResourceGroup(resourceGroupName)
+	if err != nil {
+		return false, errors.Wrapf(err, "error listing images by resourceGroup: %s", resourceGroupName)
+	}
+	for _, image := range images {
+		if image == fmt.Sprintf("%s-gen2", infraID) {
```

Comment:
Do we need to check the whole string, or would `HasSuffix(image, "-gen2")` suffice?

Not that this is bad, necessarily -- but it does assume more about the specific string format of the image name (generated by... the installer?)

Reply:
I've updated the check to test if any images found in the installer created, default resourceGroup have an `.ImageProperties.HyperVGeneration` of V2.